VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsAdapter.java

```diff
@@ -30,13 +29,12 @@
     @ColorInt
     private final int edx_utility_success_dark;
 
-    @Inject
-    public DiscussionPostsAdapter(Context context, IEdxEnvironment environment) {
-        super(context, R.layout.row_discussion_thread, environment);
-        edx_brand_primary_base = context.getResources().getColor(R.color.edx_brand_primary_base);
-        edx_grayscale_neutral_light = context.getResources().getColor(R.color.edx_grayscale_neutral_light);
-        edx_brand_secondary_dark = context.getResources().getColor(R.color.edx_brand_secondary_dark);
-        edx_utility_success_dark = context.getResources().getColor(R.color.edx_utility_success_dark);
+    public DiscussionPostsAdapter(Fragment fragment, IEdxEnvironment environment) {
+        super(fragment.getContext(), R.layout.row_discussion_thread, environment);
```

Comment:
> retaining local variable references to them is fine for now, since they'll be replaced by new references on the next view creation iteration.

That's true, but it still means the old activity is forced to remain in memory while the new activity is being created to replace it. If there are any references to views, or the previous activity context, they should be removed explicitly in `onDestroyView()`